packages/frontend-2/components/settings/Dialog.vue

```diff
@@ -77,6 +77,7 @@
           !isMobile && 'simple-scrollbar overflow-y-auto flex-1'
         ]"
         :user="user"
+        :workspace-id="targetWorkspaceId"
```

Comment:
This dynamic component & property binding is not very nice, cause there's no type checking - you can forget to add a prop or misspell it, and there's not gonna be any TS errors here. While you don't have to change this now, ideally the `menuItemConfig` type should be adjusted so that you also specify the props for the component there and then bind them in the template using `v-bind="props"`. On top of that, there should be type checking that checks that the props you've provided for each component match the prop type signature for the component.

It would probably mean that MenuItem has to be some kind of generic type like:
```
type MenuItem<Component extends PropAnyComponent> = {
  title: string
  component: Component
  props?: InstanceType<Component>['$props']
}

```

Reply:
That's a good point, it will probably be needed at some point as this might grow, different settings will require different props, I will make sure to adjust it